cmd/cmd.go

```diff
@@ -0,0 +1,176 @@
+// Package cmd - package
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github/Asac2142/go-cli-app/file"
+	"github/Asac2142/go-cli-app/task"
+)
+
+const (
+	add            string = "add"
+	update         string = "update"
+	deleteTask     string = "delete"
+	markInProgress string = "mark-in-progress"
+	markDone       string = "mark-done"
+	list           string = "list"
+)
+
+// HandleTrackerCLI - Handles user's input
+func HandleTrackerCLI(f *file.File[task.TContent]) {
+	task := task.NewTask(f)
+	args := os.Args[1:]
+
+	if len(args) == 0 {
+		err := errors.New("Invalid input")
+		printError(err)
+	}
+
+	switch args[0] {
+	case add:
+		err := addTask(&args, task)
+		printError(err)
+	case update:
+		err := updateTask(&args, task)
+		printError(err)
+	case deleteTask:
+		err := removeTask(&args, task)
+		printError(err)
+	case markInProgress:
+		err := updateStatus(&args, markInProgress, task)
+		printError(err)
+	case markDone:
+		err := updateStatus(&args, markDone, task)
+		printError(err)
+	case list:
+		err := listTasks(&args, task)
+		printError(err)
+	default:
+		err := errors.New("Invalid task command")
+		printError(err)
+	}
+}
+
+func addTask(args *[]string, t *task.Task) error {
+	if len(*args) == 1 {
+		return errors.New("add task requires a description")
+	}
+
+	desc := (*args)[1]
+	added, err := t.Add(desc)
+
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Task added successfully (ID: %d)\n", added.ID)
+
+	return nil
+}
+
+func updateTask(args *[]string, t *task.Task) error {
+	if len(*args) != 3 {
+		return errors.New("update task requires an id & description")
+	}
+
+	strID := (*args)[1]
+	id, err := strconv.Atoi(strID)
+	if err != nil {
+		return errors.New("invalid id provided")
+	}
+
+	dsc := (*args)[2]
+	return t.Update(id, dsc)
+}
+
+func removeTask(args *[]string, t *task.Task) error {
+	if len(*args) != 2 {
+		return errors.New("delete task requires an id")
+	}
+
+	asString := (*args)[1]
+	id, err := strconv.Atoi(asString)
+	if err != nil {
+		return errors.New("invalid id provided")
+	}
+
+	return t.Delete(id)
+}
+
+func updateStatus(args *[]string, status string, t *task.Task) error {
+	if len(*args) != 2 {
+		return errors.New("mark-in-progress task requires an id")
+	}
+
+	asString := (*args)[1]
+	id, err := strconv.Atoi(asString)
+	if err != nil {
+		return errors.New("invalid task id provided")
+	}
+
+	var s task.Status
+
+	if status == markInProgress {
```

Comment:
here you should use switch with default branch:

```
switch status {
case markInProgress:
    // ...
case markDone:
    // ...
default:
    return errors.New("invalid status provided")
}
```

I haven't seen this variables and constants, but I would considered to redesign it so that you don't need to assign pretty much the same as it was before.
I mean if status == inProgress { s = task.InProgress }
you should have this values equal and here only check if provided values exists and fits app logic.